### ISSUE TITLE
redux-persist failed to create sync storage. falling back to memory storage.

### ISSUE THREAD

**Issue by zhangwei900808** [zeit/next.js#9247]
I used with-redux-persist and run it. the terminal response below it
```
yarn dev
yarn run v1.19.1
$ next -p 3003
[ wait ]  starting the development server ...
[ info ]  waiting on http://localhost:3003 ...
[ ready ] compiled successfully - ready on http://localhost:3003
[ wait ]  compiling ...
[ ready ] compiled successfully - ready on http://localhost:3003
[ event ] build page: /
[ wait ]  compiling ...
[ ready ] compiled successfully - ready on http://localhost:3003
redux-persist failed to create sync storage. falling back to memory storage.
Connect(Index) returned an empty object from `getInitialProps`. This de-optimizes and prevents automatic static optimization. https://err.sh/zeit/next.js/empty-object-getInitialProps
Connect(Index) returned an empty object from `getInitialProps`. This de-optimizes and prevents automatic static optimization. https://err.sh/zeit/next.js/empty-object-getInitialProps
```

**Comment by jamsch** [comment]
Wouldn't you want to run redux-persist only on the client? The server renderer wouldn't be running a fully-fledged browser with sync/asyncstorage so you'd probably need to add an if statement around the persist middleware to make sure that the client is only running it.

**Comment by timneutkens** [comment]
Please follow the issue template.

https://github.com/zeit/next.js/issues/new/choose

https://github.com/zeit/next.js/issues/new?template=8.Question_about_next.md

```
# Question about Next.js

GitHub Issues are reserved for Bug reports and Feature requests. The best place to get your question answered is to post it on https://spectrum.chat/next-js.
```

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
